### PR TITLE
example-await-worker

### PR DESCRIPTION
**Pull request by krystian50**

### CODE REVIEW

File: src/store/utils.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import {
+  WorkerResponseParams,
+  CommandWorkerRequestParams,
+  WorkerRequestParams,
+  Commands,
+} from "@/packages/web-worker/types";
+
+const RESPONSE_WAIT_TIMEOUT = 5000;
```
Comment (wkwiatek) [inline src/store/utils.ts:8]: Response timeout, although generally is a good idea, I'd put a much larger number in here (like `60_000`). 

Right now the issue is that e.g. loading takes time as it has fetching wasm file inside so depending on server + network conditions can result in much longer wait time. 

Other remark is that after timeout we should get the UI back to the state it was initially. In the case above it stucks on spinning circles.

**Review by wkwiatek — COMMENTED** [review]
Feels like a solid cleanup!

Would love to discuss a few points mentioned in comments before getting it merged.

File: src/components/ProgramLoader/Assembly.tsx [hunk 2]
```diff
@@ -188,6 +188,8 @@ export const Assembly = ({
       >
         <CodeMirror
           autoFocus
+          style={{ height: "100%" }}
```
[the remark this answers is not in the document]
Comment (krystian50) [reply src/components/ProgramLoader/Assembly.tsx:191]: prop height is responsible for the content inside. You set the styles on wrapper by classname or styles. Changed to tailwind h-full

File: src/packages/web-worker/command-handlers/load.ts [hunk 3]
```diff
@@ -37,6 +37,8 @@ const load = async (args: LoadParams): Promise<PvmApiInterface | null> => {
 };
 
 export const runLoad = async (args: LoadParams): Promise<LoadResponse> => {
+  // return { pvm: null, status: CommandStatus.ERROR, error: new Error("Unknown PVM type") };
```
[the remark this answers is not in the document]
Comment (krystian50) [reply src/packages/web-worker/command-handlers/load.ts:40]: Si. Only for testing purpose.

File: src/packages/web-worker/utils.ts [hunk 4]
```diff
@@ -82,11 +82,11 @@ export async function loadArrayBufferAsWasm(bytes: ArrayBuffer, lang?: Supported
 
 export function getMemoryPage(pageNumber: number, pvm: PvmApiInterface | null) {
   if (!pvm) {
-    return [];
+    return new Uint8Array();
   }
 
   if (isInternalPvm(pvm)) {
-    return pvm.getMemoryPage(pageNumber) || [];
+    return pvm.getMemoryPage(pageNumber) || new Uint8Array();
```
[the remark this answers is not in the document]
Comment (krystian50) [reply src/packages/web-worker/utils.ts:89]: Sounds correct, but this is what typeberry returns` Uint8Array | null ` and I don't really know in which case they can return null

File: src/store/utils.ts (hunk 1 above)
Comment (krystian50) [reply src/store/utils.ts]: I increased timeout to 60s. We can have a longer timeout for commands like load. Others may be shorter. 60s for each command may confuse user and we can predict how much time should take specific commands. I set timeout as a promise reject to handle it similar to `commandStatus: 'error"` and now when I think about it maybe it should be `commandStatus: 'error"` with a timeout error. In such case handling would be similar to standard error from worker.

**Pull request merged**